gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -332,6 +365,54 @@ private void showOfflineAreas() {
     viewModel.showOfflineAreas();
   }
 
+  private void onShowAddPolygonDialogRequest(Point point) {
+    if (vertices.contains(point)) {
+      mapContainerViewModel.updatePolygonDrawing(PolygonDrawing.COMPLETED);
```

Comment:
Technically the polygon is only completed when the user clicked the checkmark to finish. In this case, I would say the polygon drawing state is something like "LOOP_CLOSED".